transformers/senml/transformer.go

```diff
@@ -20,20 +27,19 @@ var formats = map[string]senml.Format{
 	CBOR: senml.CBOR,
 }
 
-type transformer struct{}
+type transformer struct {
+	format senml.Format
+}
 
 // New returns transformer service implementation for SenML messages.
-func New() transformers.Transformer {
-	return transformer{}
+func New(contentType string) transformers.Transformer {
+	return transformer{
+		format: formats[contentType],
+	}
 }
 
 func (n transformer) Transform(msg broker.Message) (interface{}, error) {
```

Comment:
You can rename `func (n transformer) Transform...` to `func (t transformer) Transform...`. Weird receiver name `n` is a legacy from `Normalizer`.